awswrangler/oracle.py

```diff
@@ -0,0 +1,427 @@
+    )
+
+
+@_check_for_cx_Oracle
+def read_sql_table(
+    table: str,
+    con: "cx_Oracle.Connection",
+    schema: Optional[str] = None,
+    index_col: Optional[Union[str, List[str]]] = None,
+    params: Optional[Union[List[Any], Tuple[Any, ...], Dict[Any, Any]]] = None,
+    chunksize: Optional[int] = None,
+    dtype: Optional[Dict[str, pa.DataType]] = None,
+    safe: bool = True,
+    timestamp_as_object: bool = False,
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    """Return a DataFrame corresponding the table.
+
+    Parameters
+    ----------
+    table : str
+        Table name.
+    con : cx_Oracle.Connection
+        Use cx_Oracle.connect() to use credentials directly or wr.oracle.connect() to fetch it from the Glue Catalog.
+    schema : str, optional
+        Name of SQL schema in database to query (if database flavor supports this).
+        Uses default schema if None (default).
+    index_col : Union[str, List[str]], optional
+        Column(s) to set as index(MultiIndex).
+    params :  Union[List, Tuple, Dict], optional
+        List of parameters to pass to execute method.
+        The syntax used to pass parameters is database driver dependent.
+        Check your database driver documentation for which of the five syntax styles,
+        described in PEP 249’s paramstyle, is supported.
+    chunksize : int, optional
+        If specified, return an iterator where chunksize is the number of rows to include in each chunk.
+    dtype : Dict[str, pyarrow.DataType], optional
+        Specifying the datatype for columns.
+        The keys should be the column names and the values should be the PyArrow types.
+    safe : bool
+        Check for overflows or other unsafe data type conversions.
+    timestamp_as_object : bool
+        Cast non-nanosecond timestamps (np.datetime64) to objects.
+
+    Returns
+    -------
+    Union[pandas.DataFrame, Iterator[pandas.DataFrame]]
+        Result as Pandas DataFrame(s).
+
+    Examples
+    --------
+    Reading from Oracle Database using a Glue Catalog Connections
+
+    >>> import awswrangler as wr
+    >>> con = wr.oracle.connect(connection="MY_GLUE_CONNECTION")
+    >>> df = wr.oracle.read_sql_table(
+    ...     table="my_table",
+    ...     schema="test",
+    ...     con=con
+    ... )
+    >>> con.close()
+    """
+    table_identifier = _get_table_identifier(schema, table)
+    sql: str = f"SELECT * FROM {table_identifier}"
+    return read_sql_query(
+        sql=sql,
+        con=con,
+        index_col=index_col,
+        params=params,
+        chunksize=chunksize,
+        dtype=dtype,
+        safe=safe,
+        timestamp_as_object=timestamp_as_object,
+    )
+
+
+@_check_for_cx_Oracle
+@apply_configs
+def to_sql(
+    df: pd.DataFrame,
+    con: "cx_Oracle.Connection",
+    table: str,
+    schema: str,
+    mode: str = "append",
+    index: bool = False,
+    dtype: Optional[Dict[str, str]] = None,
+    varchar_lengths: Optional[Dict[str, int]] = None,
```

Comment:
Is this required as a separate parameter? Could it be smth like `dtype={"col0": "VARCHAR(30)"}`?

Reply:
Thx agreed